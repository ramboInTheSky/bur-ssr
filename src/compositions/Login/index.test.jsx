import { shallow } from "enzyme";
import toJson from 'enzyme-to-json';
import * as React from "react";
import Loader from "react-loader";
import { LoginComponent } from ".";

describe("Testing login component", () => {
  let props = {
    history: {
      push: jest.fn()
    },
    login: jest.fn()
  };

  beforeEach(() => {
    props = {
      history: {
        push: jest.fn()
      },
      login: jest.fn()
    };
  });

  it("renders correctly ", () => {
    expect(shallow(<LoginComponent {...props} />)).toMatchSnapshot();
  });

  it("Loader shows up if loading is set to false", () => {
    const wrapper = shallow(<LoginComponent {...props} />);
    wrapper.setState({
      loading: true
    });
    expect(wrapper.find(Loader)).toHaveLength(1);
    expect(toJson(wrapper)).toMatchSnapshot();
  });



});
