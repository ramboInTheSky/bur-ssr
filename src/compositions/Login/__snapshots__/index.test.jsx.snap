// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing login component Loader shows up if loading is set to false 1`] = `
"<div style={{...}}>
  <div className=\\"form-container login\\" style={{...}}>
    <h1>
      LogIn
    </h1>
    <Component loaded={false} options={{...}} component=\\"div\\" loadedClassName=\\"loadedContent\\" parentClassName=\\"loader\\">
      <Formik initialValues={{...}} validationSchema={{...}} onSubmit={[Function: onSubmit]} validateOnChange={true} validateOnBlur={true} isInitialValid={false} enableReinitialize={false}>
        [function]
      </Formik>
    </Component>
  </div>
</div>"
`;

exports[`Testing login component Text shows up when the user is typing 1`] = `
"<div style={{...}}>
  <div className=\\"form-container login\\" style={{...}}>
    <h1>
      LogIn
    </h1>
    <Component loaded={true} options={{...}} component=\\"div\\" loadedClassName=\\"loadedContent\\" parentClassName=\\"loader\\">
      <Formik initialValues={{...}} validationSchema={{...}} onSubmit={[Function: onSubmit]} validateOnChange={true} validateOnBlur={true} isInitialValid={false} enableReinitialize={false}>
        [function]
      </Formik>
    </Component>
  </div>
</div>"
`;

exports[`Testing login component renders correctly  1`] = `
"<div style={{...}}>
  <div className=\\"form-container login\\" style={{...}}>
    <h1>
      LogIn
    </h1>
    <Component loaded={true} options={{...}} component=\\"div\\" loadedClassName=\\"loadedContent\\" parentClassName=\\"loader\\">
      <Formik initialValues={{...}} validationSchema={{...}} onSubmit={[Function: onSubmit]} validateOnChange={true} validateOnBlur={true} isInitialValid={false} enableReinitialize={false}>
        [function]
      </Formik>
    </Component>
  </div>
</div>"
`;
